src/ThemeSelector.js

```diff
@@ -1,42 +1,153 @@
+    };
+
+    renderThemeOptions = () => {
+        const themeGroup = [];
+        for (const theme of this.themeOptions) {
+            const isChecked = this.props.currentTheme === theme;
+            themeGroup.push(
+                <div
+                    role='radio'
+                    aria-checked={isChecked}
+                    className={`ThemeSelector__option ${theme}`}
+                    key={`ThemeSelector__option-${theme}`}
+                    data-theme={theme}
+                    tabIndex={isChecked ? 0 : -1}
+                    onClick={this.handleOnSelect}
+                    onKeyDown={this.handleKeyDown}>
+                    <input className='ThemeSelector__option-radio' type='radio' name='theme-option' value={theme}
+                        aria-hidden={true}
+                        checked={isChecked}
+                        readOnly={true}
+                        tabIndex={-1}/>
+                    <FormattedMessage id={`ThemeSelector.option.${theme}`} />
+                </div>
+            );
+        }
+        return themeGroup;
+    }
+
+    componentDidUpdate(prevProps: ThemeSelectorProps) {
+        // When the modal first open up, remember the theme at that time
+        if (prevProps.show !== this.props.show && this.props.show) {
+            this.setState({
+                selectedTheme: this.props.currentTheme
+            });
+        }
+    }
+
     render() {
         return (
-            <DropdownButton className='ThemeSelector' onSelect={this.props.onSelect} title={`${this.props.intl.formatMessage({id:'ThemeSelector.theme'})}`}>
-                {
-                    /* $FlowFixMe
-                        Cannot get Dropdown.Item because property Item is missing in statics of Dropdown
-                    */
-                    <Dropdown.Item eventKey='mixed'>{this.props.intl.formatMessage({id:'ThemeSelector.theme.mixed'})}</Dropdown.Item>
-                }
-                {
-                    // $FlowFixMe
-                    <Dropdown.Item eventKey='light'>{this.props.intl.formatMessage({id:'ThemeSelector.theme.light'})}</Dropdown.Item>
-                }
-
-                { // $FlowFixMe
-                    <Dropdown.Item eventKey='dark'>{this.props.intl.formatMessage({id:'ThemeSelector.theme.dark'})}</Dropdown.Item>
-                }
-                { // $FlowFixMe
-                    <Dropdown.Item eventKey='gray'>{this.props.intl.formatMessage({id:'ThemeSelector.theme.gray'})}</Dropdown.Item>
-                }
-                { // $FlowFixMe
-                    <Dropdown.Item eventKey='contrast'>{this.props.intl.formatMessage({id:'ThemeSelector.theme.contrast'})}</Dropdown.Item>
-                }
-            </DropdownButton>
+            <ModalWithFooter
+                show={this.props.show}
+                focusOnOpenSelector={`.ThemeSelector__option.${this.props.currentTheme}`}
+                focusOnCloseSelector='.IconButton.themeSelector'
```

Comment:
The close selector isn't working at the moment, that button has `class="IconButton App__header-themeSelectorIcon"`.   So you'd need `.IconButton.App__header-themeSelectorIcon` instead.

Reply:
Thanks @the-t-in-rtf, nice catch 👍